vpr/src/place/noc_place_utils.h

```diff
@@ -0,0 +1,375 @@
+ * The link bandwidth usages are either incremeneted or decremented by the 
+ * bandwidth of the traffic flow. If the traffic flow route is being deleted,
+ * then the link bandwidth needs to be decremented. If the traffic flow
+ * route has just been added then the link badnwidth needs to be incremented.
+ * This function needs to be called everytime a traffic flow has been newly
+ * routed.
+ * 
+ * @param traffic_flow_route The routed path for a traffic flow. This
+ * contains a collection of links in the NoC.
+ * @param noc_model Contains all the links and rotuters within the NoC. Used
+ * to update link information.
+ * @param how_to_update_links Determines how the bandwidths of links found
+ * in the traffic flow route are updated.
+ * @param traffic_flow_bandwidth The bandwidth of a traffic flow. This will
+ * be used to update bandwidth usage of the links.
+ */
+void update_traffic_flow_link_usage(const std::vector<NocLinkId>& traffic_flow_route, NocStorage& noc_model, link_usage_update_state how_to_update_links, double traffic_flow_bandwidth);
+
+/**
+ * @brief Goes through all the traffic flows associated to a moved
+ * logical router cluster block (a traffic flow is associated to a router if
+ * the router is either a source or sink router of the traffic flow) and
+ * re-routes them. The new routes are stored and the NoC cost is updated
+ * to reflect the moved logical router cluster block.
+ * 
+ * @param moved_router_block_id The logical router cluster block that was moved
+ * to a new location during placement.
+ * @param noc_traffic_flows_storage Contains all the traffic flow information
+ * within the NoC. Used to get the traffic flows associated to logical router
+ * blocks.
+ * @param noc_model Contains all the links and rotuters within the NoC. Used
+ * to route traffic flows within the NoC.  
+ * @param noc_flows_router The packet routing algorithm used to route traffic
+ * flows within the NoC.
+ * @param placed_cluster_block_locations A datastructure that identifies the
+ * placed grid locations of all cluster blocks.
+ * @param updated_traffic_flows Keeps track of traffic flows that have been
+ * re-routed. Used to prevent re-rouitng the same traffic flow multiple times.
+ */
+void re_route_associated_traffic_flows(ClusterBlockId moved_router_block_id, NocTrafficFlows& noc_traffic_flows_storage, NocStorage& noc_model, NocRouting* noc_flows_router, const vtr::vector_map<ClusterBlockId, t_block_loc>& placed_cluster_block_locations, std::unordered_set<NocTrafficFlowId>& updated_traffic_flows, int& number_of_affected_traffic_flows);
+
+/**
+ * @brief Used to re-route all the traffic flows associated to logical
+ * router blocks that were supposed to be moved during placement but are
+ * back to their original positions.
+ * 
+ * @param blocks_affected Contains all the blocks that were moved in
+ * the current placement iteration. THis includes the cluster ids of
```

Comment:
THis -> This

Reply:
Either here or in .cpp, mention that this is done by restoring a saved version of the flows.

Reply:
Looks like we are calling the routing function again to find the original route of the traffic flow before the blocks were swapped. I think I did this because I didn't want to have a global data-structure which I would then use to store new paths and then copy them over if the move was accepted and then deleting the paths for the next placement iteration. We can discuss this further but for now I think we should leave it as is and then make the change in the future.